### PR TITLE
E0283 diagnostics

### PR DESCRIPTION
**Pull request by therustmonk**
Example for E0283 error

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -1587,6 +1587,52 @@ This will fail because the compiler does not know which instance of `Foo` to
 call `bar` on. Change `Foo::bar()` to `Foo::<T>::bar()` to resolve the error.
 "##,
 
+E0283: r##"
+This error raises when compiler hasn't enough information about type to
```
Comment (nagisa) [inline src/librustc/diagnostics.rs]: s/raises/occurs/ 

about the type

File: src/librustc/diagnostics.rs [hunk 2]
```diff
@@ -1587,6 +1587,52 @@ This will fail because the compiler does not know which instance of `Foo` to
...
+choose implementation.
```
Comment (nagisa) [inline src/librustc/diagnostics.rs]: choose an implementation

File: src/librustc/diagnostics.rs [hunk 3]
```diff
@@ -1587,6 +1587,52 @@ This will fail because the compiler does not know which instance of `Foo` to
 call `bar` on. Change `Foo::bar()` to `Foo::<T>::bar()` to resolve the error.
 "##,
 
+E0283: r##"
```
Comment (nagisa) [inline src/librustc/diagnostics.rs]: You need to remove the error code from that other error list that lists all the error codes without explanations.

File: src/librustc/diagnostics.rs [hunk 4]
```diff
@@ -1587,6 +1587,52 @@ This will fail because the compiler does not know which instance of `Foo` to
...
+trait Outer {
+    fn new() -> Inner;
+}
+impl Outer for Inner {
+    fn new() -> Inner {
+        Inner {out:true}
+    }
+}
+fn main() {
+    let ext: Inner = Outer::new(); // error, Inner expected in any way
```
Comment (nagisa) [inline src/librustc/diagnostics.rs]: I think this should actually compile, or it does fail to compile for another reason than you try to explain.

**Comment by DenisKolodin** [comment]
Thank you for the comments. I'll fix them locally, at first, and send updates later.

**Pull request closed without merging**